src/useFetchye.js

```diff
@@ -14,48 +14,48 @@
  * permissions and limitations under the License.
  */
 
-import { useContext, useEffect, useRef } from 'react';
+import { useEffect, useRef } from 'react';
 import { runAsync } from './runAsync';
-import { defaultFetcher } from './defaultFetcher';
 import { computeKey } from './computeKey';
-import { isLoading } from './isLoading';
-import { FetchyeContext } from './FetchyeContext';
-import { getCacheByKey } from './getCacheByKey';
+import {
+  isLoading,
+  getData,
+  getError,
+} from './queryHelpers';
+import { useFetchyeContext } from './useFetchyeContext';
 import { defaultMapOptionsToKey } from './defaultMapOptionsToKey';
 
 export const useFetchye = (
   key,
   { mapOptionsToKey = (options) => options, ...options } = { },
-  fetcher = defaultFetcher) => {
+  fetcher) => {
   const {
-    useFetchyeSelector, dispatch, fetchClient,
-  } = useContext(FetchyeContext);
-  const cache = useFetchyeSelector();
+    defaultFetcher, useFetchyeSelector, dispatch, fetchClient,
+  } = useFetchyeContext();
+  const selectedFetcher = typeof fetcher === 'function' ? fetcher : defaultFetcher;
   const computedKey = computeKey(key, defaultMapOptionsToKey(mapOptionsToKey(options)));
-  const { data, loading, error } = getCacheByKey(cache, computedKey);
-  const isFirstRender = useRef(!data);
+  const { data, loading, error } = useFetchyeSelector(computedKey.hash);
+  const numOfRenders = useRef(0);
+  numOfRenders.current += 1;
   useEffect(() => {
     if (options.lazy || !computedKey) {
       return;
     }
-    if (isFirstRender.current !== false) {
-      isFirstRender.current = false;
-    }
     if (!data && !error && !loading) {
       (async () => {
         await runAsync({
-          dispatch, computedKey, fetcher, fetchClient, options,
+          dispatch, computedKey, fetcher: selectedFetcher, fetchClient, options,
```

Comment:
What's the difference between `fetcher` and `fetchClient`? That has me a bit confused